File: kinotic-idl/src/main/java/org/kinotic/idl/api/schema/AsyncC3Type.java
package org.kinotic.idl.api.schema;

import lombok.*;
import lombok.experimental.Accessors;

/**
 * Represents a value that is produced asynchronously, resolving to a single {@link #valueType} result.
 * Created by Navíd Mitchell 🤪 on 7/22/26.
 */
@Getter
@Setter
@Accessors(chain = true)
@NoArgsConstructor
@AllArgsConstructor
@EqualsAndHashCode(callSuper = true)
@ToString(callSuper = true)
public class AsyncC3Type extends C3Type {

    /**
     * The {@link C3Type} of the value the asynchronous invocation resolves to
     */
    private C3Type valueType = null;

}

File: kinotic-js/workspace/packages/idl/src/api/AsyncC3Type.ts
import {C3Type} from '@/api/C3Type'

/**
 * Represents a value that is produced asynchronously, resolving to a single result.
 * Created by Navíd Mitchell 🤪 on 7/22/26.
 */
export class AsyncC3Type extends C3Type {
    /**
     * The type of the value the asynchronous invocation resolves to
     */
    public valueType: C3Type

    constructor(valueType: C3Type) {
        super('async')
        this.valueType = valueType
    }
}
